static/js/compose_ui.js

```diff
@@ -426,3 +431,44 @@ export function get_compose_click_target(e) {
     }
     return e.target;
 }
+
+export function on_compose_select_stream_update(new_value) {
+    const $stream_header_colorblock = $(".stream_selection_dropdown").find(
+        ".stream_header_colorblock",
+    );
+    stream_bar.decorate(new_value, $stream_header_colorblock, true);
+    compose.update_on_recipient_change();
+    $("#stream_message_recipient_topic").trigger("focus").trigger("select");
+}
+
+export let compose_stream_widget;
+export function initialize_compose_stream_dropdown() {
+    const streams_list = stream_data
+        .subscribed_subs()
+        .filter((stream) => stream_data.can_post_messages_in_stream(stream))
+        .map((stream) => ({
+            name: stream.name,
+            value: stream.name,
+        }));
+    const opts = {
+        widget_name: "compose_select_stream",
+        data: streams_list,
+        default_text: $t({defaultMessage: "Select a stream"}),
+        value: null, // The value is set when the compose box opens.
```

Comment:
Related to my [other comment](https://github.com/zulip/zulip/pull/23355/commits/69566f08ed6bdae6d3ee738172e1c211a5ba72d6#r1063917832), here it's split into `default_text` and `value`, but once it's rendered I couldn't figure out how to get the pure value again.